sklearn/linear_model/least_angle.py

```diff
@@ -1074,6 +1074,8 @@ def fit(self, X, y):
         """
         self.fit_path = True
         X, y = check_X_y(X, y, y_numeric=True)
+        X = as_float_array(X, copy=self.copy_X)
```

Comment:
Well, if you think upcasting int32 to float64 is an issue, we should get it fixed everywhere.